core_api_c_binding/src/lib.rs

```diff
@@ -27,17 +27,21 @@ impl Logger for NullLogger {
 pub unsafe extern "C" fn holochain_new(ptr: *mut Dna) -> *mut Holochain {
     let agent = Agent::from("c_bob".to_string());
 
-    let context = Arc::new(Context::new(
+    let context = Context::new(
```

Comment:
I would rename to `maybe_context` or some other name that makes it clear its a result.